sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -54,6 +54,8 @@ trait HashJoin extends BaseJoinExec {
 
   override def outputPartitioning: Partitioning = streamedPlan.outputPartitioning
 
+  override def outputOrdering: Seq[SortOrder] = streamedPlan.outputOrdering
```

Comment:
to be future-proof, we should only do it if the join type allows us to do so. It's fragile to rely on what the join type can be in hash join.

Reply:
@cloud-fan - agree. Updated both `outputPartitioning` and  `outputOrdering` to be based on join type and build side.